File: hw/astudillo/U3/HW22Observer/Observer/src/main/java/ec/edu/espe/observer/model/Investor.java
package ec.edu.espe.observer.model;

/**
 * Concrete Observer in the Observer pattern
 * @author Gerald Astudillo
 */
public class Investor implements IInvestor {
    private String name;

    public Investor(String name) {
        this.name = name;
    }

    public void update(Stock stock, Object args) {
        System.out.println("Notified observer " + name);
        if (args instanceof String) {
            System.out.println(args);
        } else if (args instanceof Double) {
            System.out.println("The price of " + stock.getSymbol() + " changed to: " + args);
        }
    }
}

File: ws/paillacho/U3/WS35Observer/ObserverPython/src/model/investor.py
"""
Concrete Observer in the Observer pattern
@author Gerald Astudillo
"""

from .i_investor import IInvestor

class Investor(IInvestor):
    def __init__(self, name):
        self._name = name

    def update(self, stock, args):
        """Update method called when stock changes"""
        print(f"Notified observer {self._name}")
        if isinstance(args, str):
            print(args)
        elif isinstance(args, float):
            print(f"The price of {stock.get_symbol()} changed to: {args}")

    @property
    def name(self):
        return self._name
